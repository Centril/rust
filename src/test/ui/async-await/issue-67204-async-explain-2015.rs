// edition:2015

fn main() {}
fn bar() -> Result<(), ()> {
    panic!()
}

fn foo() -> Result<(), ()> {
   let x = async {
       bar()?;
   };
   Ok(())
}
