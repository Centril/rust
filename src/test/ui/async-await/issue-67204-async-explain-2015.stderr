error: `async` block not allowed on Rust 2015
  --> $DIR/issue-67204-async-explain-2015.rs:9:12
   |
LL |    let x = async {
   |            ^^^^^
   |
   = note: to use an `async` block, switch to Rust 2018
   = help: set `edition = "2018"` in `Cargo.toml`
   = note: for more on editions, read https://doc.rust-lang.org/edition-guide

error[E0277]: the `?` operator can only be used in an async block that returns `Result` or `Option` (or another type that implements `std::ops::Try`)
  --> $DIR/issue-67204-async-explain-2015.rs:10:8
   |
LL |      let x = async {
   |  __________________-
LL | |        bar()?;
   | |        ^^^^^^ cannot use the `?` operator in an async block that returns `()`
LL | |    };
   | |____- this function should return `Result` or `Option` to accept `?`
   |
   = help: the trait `std::ops::Try` is not implemented for `()`
   = note: required by `std::ops::Try::from_error`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
